Core/include/Acts/Seeding/PathSeeder.hpp

```diff
@@ -0,0 +1,185 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Seeding/ISourceLinkGrid.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+
+namespace Acts {
+
+/// @brief Seeding algorigthm that extracts
+/// the IP parameters and sorts the source links
+/// into possible track candidates
+///
+/// @tparam axis_t Type of the axis to bin
+/// the source links
+///
+template <typename axis_t>
+class PathSeeder {
+ public:
+  struct Seed {
+    ActsScalar ipP;
+    Vector3 ipDir;
+    Vector3 ipVertex;
+    std::vector<SourceLink> sourceLinks;
+
+    Seed() = delete;
+    Seed(ActsScalar ipPmag, Vector3 ipPdir, Vector3 ipPos,
+         std::vector<SourceLink> sls)
+        : ipP(ipPmag),
+          ipDir(std::move(ipPdir)),
+          ipVertex(std::move(ipPos)),
+          sourceLinks(std::move(sls)){};
+  };
+
+  using SourceLinkCalibrator =
+      Delegate<Vector3(const GeometryContext&, const SourceLink&)>;
+
+  using PathWidthLookup = Delegate<std::pair<ActsScalar, ActsScalar>(
+      const GeometryContext&, const GeometryIdentifier&)>;
+
+  using IntersectionLookup =
```

Comment:
Is this direction, position, line ? 
(does this imply that the surfaces are always perpenticular)?

Reply:
I wonder if that could be all hidden, btw.
If this became `const Vector3&, const Vector3&, const Surface&, GeometryContext&` and in case of a 
`| | | |` code you use your super-optimised implementation, but even if your detector looks `\ \ \ \` it could still work with a more complicated delegate.

Reply:
From our discussion I now understood that the scalar is the momentum magnitude, then we need the charge as well.